### PR TITLE
feat: Update neard init to specify config file type to download

### PR DESCRIPTION
**Pull request by tayfunelmas**
We change the `--download-config` parameter to take a string.
- `neard init --download-config validator` downloads the `config.json` file for validators (tracking single shard, loading memtries).
- `neard init --download-config rpc` downloads the `config.json` file for non-validator RPC nodes (tracking all shards, NOT loading memtries).
- `neard init --download-config archival` downloads the `config.json` file for non-validator archival nodes (tracking all shards, NOT loading memtries, archive field set to true).
- `neard init --download-config` (with no value) defaults to `validator` config (for backwards compatibility).

We map the flag to a specific URL containing the config type. 
The new config files per config type can be found at https://github.com/tayfunelmas/near-configs.
The new config files are already uploaded to AWS S3 with the new URLs.

NOTE: For indexer and chainsync-loadtest params, we use `rpc` config type.

Next: We will update the Near validator documentation to use the new flag value.

Tracking issue: #12070

### CODE REVIEW

**Comment by tayfunelmas** [comment]
Adding @khorolets as reviewer due to the flag change for Indexers.

File: chain/indexer/src/lib.rs [hunk 1]
```diff
@@ -43,7 +45,7 @@ pub struct InitConfigArgs {
     /// Specify a custom download URL for the records file.
     pub download_records_url: Option<String>,
     /// Download the verified NEAR config file automatically.
-    pub download_config: bool,
+    pub download_config: Option<String>,
```
Comment (marcelo-gonzalez) [inline chain/indexer/src/lib.rs:48]: nit: why not just make this field an `Option<DownloadConfigType>`? The only place this is used is in the one place in the indexer example, where we are allocating a string and writing "rpc" to it, and then calling `DownloadConfigType::from_str()` on the string we just allocated. we already know what's gonna be in the string bc we just wrote to it. you could just change indexer/example/src/configs.rs to say this instead:

```rust
if config_args.download_config {
  Some(DownloadConfigType::Rpc)
} else {
  None
}
```

Or you could even leave it as-is as a bool, and in `indexer_init_configs()` check that unchanged bool field and set it to `DownloadConfigType::Rpc` if it's true.

File: nearcore/src/config.rs [hunk 2]
```diff
@@ -792,6 +792,41 @@ fn set_block_production_delay(chain_id: &str, fast: bool, config: &mut Config) {
     }
 }
 
+/// Type of the configuration to download.
+pub enum DownloadConfigType {
+    /// Validator node configuration.
+    Validator,
+    /// Non-validator RPC node configuration.
+    Rpc,
```
Comment (marcelo-gonzalez) [inline nearcore/src/config.rs:800]: nit (not a strong opinion): maybe better style would be all uppercase? we already have that style in the `DbKind` type: https://github.com/near/nearcore/blob/ea8ec289bc188bb6fecd0390a6409240b0f1590f/core/store/src/metadata.rs#L32 and

**Review by marcelo-gonzalez — APPROVED** [review]

File: chain/indexer/src/lib.rs (hunk 1 above)
Comment (tayfunelmas) [reply chain/indexer/src/lib.rs]: I did not do it due to a dependency change, but I found a good place (near-config-utils) to put the new struct, so done with the suggestion, making the type `Option<DownloadConfigType>`.

**Review by kobayurii — APPROVED** [review]
Looks great! Thank you!

**Pull request merged**